### PR TITLE
Graphics API option

### LINKED ISSUE #24: Add option for enabling AF and AA if the Win 10 compat patch is enabled
Main GitHub repository issue: https://github.com/BC46/freelancer-hd-edition/issues/53

Create a new setup page if the Win 10 compat option has been checked (enabled). This new setup page should have 2 new options that represent Anisotropic Filtering and Anti-aliasing.

EXE/dgvoodoo.conf is dgVoodoo's configuration file, which is binary. Here are the offsets that should be updateable:

Refresh rate:
Offset: 6D, 
Datatype: Int16

Antialiasing (MSAA):
Offset: 6A, 
Datatype: Byte
Possible values:
00 (Off),
02 (2x),
04 (4x),
08 (8x),

Anisotropic Filtering:
Offset: 86, 
Datatype: Byte
Possible values:
00 (Off),
02 (2x),
04 (4x),
08 (8x),
10 (16x)

Editing binary files in Inno: https://stackoverflow.com/questions/53309903/how-to-change-certain-bytes-in-huge-file-in-inno-setup

### LINKED ISSUE #69: Change File version from 0.0.0.0
image

### PR DESCRIPTION
**Pull request by BC46**
This pull request adds the following:

1. Converted Windows 10 compatibility patch to a Graphics API option with 4 choices.
2. Add custom pages for dgVoodoo and DxWrapper.
3. Exposed Anti-Aliasing and Anisotropic filtering settings.
4. Optional ReShade config.
5. Removed unnecessary event handler overrides.
6. Hex edits in binary files.
7. Grabs the user's refresh rate from the Win32 api.

Closes #69 
Closes #24 

When this pull request has been merged, the following issues from the main repo may be closed:
https://github.com/BC46/freelancer-hd-edition/issues/60
https://github.com/BC46/freelancer-hd-edition/issues/53
https://github.com/BC46/freelancer-hd-edition/issues/77

### CODE REVIEW

File: utilities.iss [hunk 1]
```diff
@@ -71,6 +71,65 @@ begin
   end;
 end;
 
+// Gets the device contextt
+function GetDC(HWND: DWord): DWord; external 'GetDC@user32.dll stdcall';
+
+// Retrieve information about a device
+function GetDeviceCaps (hDC, nIndex: Integer): Integer;
```
Comment (ollieraikkonen) [inline utilities.iss:78]: Needs testing on WINE probably

File: utilities.iss [hunk 2]
```diff
@@ -71,6 +71,65 @@ begin
   end;
 end;
 
+// Gets the device contextt
+function GetDC(HWND: DWord): DWord; external 'GetDC@user32.dll stdcall';
```
Comment (ollieraikkonen) [inline utilities.iss:75]: Needs testing on WINE probably

**Review by ollieraikkonen — COMMENTED** [review]
Had a quick look at the code, looks ok. If it works (also on WINE) then I'm happy for this to be merged

File: utilities.iss [hunk 3]
```diff
@@ -71,6 +71,65 @@ begin
   end;
 end;
 
+// Gets the device contextt
```
Comment (BC46) [inline utilities.iss:74]:
```suggestion
// Gets the device context
```

File: utilities.iss (hunk 2 above)
Comment (BC46) [reply utilities.iss:75]: I'm assuming it'll work fine since Wine should correctly translate the API call. Of course it should still be tested.

**Comment by BC46** [comment]
Everything worked well. Except for the binary edits on Linux because Wine is missing the implementation for a function that the installer requires. I've added a short message for Wine users with an explanation on how this can be mitigated.

**Pull request merged**